docusaurus/docs/internal/code-base-works/internal-dashboard-api.md

```diff
@@ -0,0 +1,153 @@
+4. **Instantiation**: A Vue plugin (`internalApiPlugin`) automatically scans `shell/plugins/internal-api/`, imports each `.api.ts` file, creates a new instance of its default export, and injects it onto the Vue prototype as `$myNewApi`.
+
+---
+
+## Steps to Create & Use a New API
+
+1. **Create a File**  
+   Within `shell/plugins/internal-api/` (or its subdirectories), create a `.api.ts` file. For example:
+
+   `shell/plugins/internal-api/my-api.api.ts`
+
+2. **Export a Default Class**  
+   Your class should extend the provided `BaseApi` class. This ensures that your API has access to common functionality (like the Vuex store) without reimplementing it for every API. The class must have:
+
+   - A constructor that calls the `BaseApi` constructor with the Vuex.Store instance.
+   - A static `apiName` method defining how it will be injected.
+   - Custom methods or logic you need to expose to components.
+
+   Example:
+
+   ```ts
+   // shell/plugins/internal-api/my-api.api.ts
+   import BaseApi from "@shell/plugins/internal-api/shared/base-api";
+
+   export default class MyApi extends BaseApi {
+     // Name under which this API is injected => this.$myApi
+     // The `$` prefix will be automatically added when the API is injected
+     static apiName() {
+       return "myApi";
+     }
+
+     public doSomething() {
+       console.log("Hello from MyApi");
+     }
+   }
+   ```
+
+3. **Usage in Vue Components**  
+   Once the file is saved, the `internalApiPlugin` (as shown in your `install-plugins.js`) will automatically discover your API file (e.g. `my-api.api.ts`), instantiate it with the Vuex store, and inject an instance of `MyApi` onto the Vue prototype. This means that:
+
+   - In an Options API component, you can access your API via this.$myApi.
+   - In a Composition API component, you can access it using getCurrentInstance() (or via other injection methods) to retrieve the global properties.
+
+### Options API Example
+
+```vue
+<template>
+  <div>Options API Component: Check the console for API output.</div>
+</template>
+
+<script>
+export default {
+  name: "OptionsComponent",
+  mounted() {
+    // Using the injected API from the Vue prototype
+    this.$myApi.doSomething();
+  },
+};
+</script>
+```
+
+### Composition API Example
+
+```vue
+<template>
+  <div>Composition API Component: Check the console for API output.</div>
+</template>
+
+<script setup>
+import { getCurrentInstance, onMounted } from "vue";
+
+// getCurrentInstance() gives you access to the component instance,
+// which contains the proxy with all global properties including injected APIs.
+const { proxy } = getCurrentInstance();
+
+onMounted(() => {
+  // Access the API via the proxy
+  proxy.$myApi.doSomething();
+});
+</script>
+```
+
+---
+
+## TypeScript Augmentation (Optional)
```

Comment:
I think you could also create a `myApi.d.ts` file in the plugins/internal-api folder, e.g.:

```
import type MyApi from "@shell/plugins/internal-api/my-api.api";

/* eslint-disable */
export {};

declare module 'vue' {
  interface ComponentCustomProperties {
    $myApi: MyApi;
  }
}

```

This certainly works within the shell, for an extension using as a library, it may be that we need to add ` // Reference` into the vue.shim file.

Just wondering if there is a way someone can declare and add an api definition just by creating a `.d.ts` file

Reply:
That's a good point. I tested this with the example described in the documentation and it works well at least in the Dashboard. I'll have to confirm if it works in extensions without a `Reference` added to the vue.shim. But for now I have updated the documentation to reflect this.

Reply:
Thanks